db/range_del_aggregator.cc

```diff
@@ -34,7 +34,6 @@ TruncatedRangeDelIterator::TruncatedRangeDelIterator(
     pinned_bounds_.emplace_back();
     auto& parsed_smallest = pinned_bounds_.back();
     Status pikStatus = ParseInternalKey(smallest->Encode(), &parsed_smallest);
-    pikStatus.PermitUncheckedError();
```

Comment:
Here I believe PermitUncheckedError() is used to work-around an incorrect handling. `assert(pikStatus.ok())` below indicates that the correct way is to handle this failure is to invalidate the iterator. Now with the change, we would loss the visibility to this error.

Reply:
Right, `PermitUncheckedError()` was added here to indicate we do not have correct error handling yet.